android-agent/build.gradle.kts

```diff
@@ -9,7 +9,6 @@ android {
 
 dependencies {
     api(project(":agent-api"))
-    api(platform(libs.opentelemetry.platform.alpha)) // Required for sonatype publishing
```

Comment:
I don't think we can merge this, due to the problems that I've now encountered multiple times during the release process. Namely, sonatype does some additional validations that will fail if this platform dependency isn't there.